src/acp/rule.ts

```diff
@@ -233,3 +249,268 @@ export function setForbiddenRuleOnPolicy(policy: Policy, rule: Rule): Policy {
+ * ```
+ *
+ * Initialise a new empty [[SolidDataset]] to store [[Rule]]'s in.
+ */
+export const createRuleDataset = createSolidDataset;
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Initialise a new, empty [[Rule]].
+ *
+ * @param url URL that identifies this [[Rule]].
+ */
+export function createRule(url: Url | UrlString): Rule {
+  const stringUrl = internal_toIriString(url);
+  let ruleThing = createThing({ url: stringUrl });
+  ruleThing = setUrl(ruleThing, rdf.type, acp.Rule);
+  return ruleThing;
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Get the [[Rule]] with the given URL from an [[RuleDataset]].
+ *
+ * @param ruleResource The Resource that contains the given [[Rule]].
+ * @param url URL that identifies this [[Rule]].
+ * @returns The requested [[Rule]], if it exists, or `null` if it does not.
+ */
+export function getRule(
+  ruleResource: RuleDataset,
+  url: Url | UrlString
+): Rule | null {
+  const foundThing = getThing(ruleResource, url);
+  if (foundThing === null || getUrl(foundThing, rdf.type) !== acp.Rule) {
+    return null;
+  }
+  return foundThing;
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * List all the agents a [[Rule]] applies **directly** to. This will not include agents
+ * that are part of a group the [[Rule]] applies to, nor will it include specific agent
+ * classes, such as authenticated or public agents.
+ *
+ * @param rule The rule from which agents are read.
+ * @returns A list of the WebIDs of agents included in the rule.
+ * @since Unreleased
+ */
+export function getAgentForRuleAll(rule: Rule): WebId[] {
+  return getIriAll(rule, acp.agent).filter(
+    (agent: WebId) =>
+      agent !== acp.PublicAgent && agent !== acp.AuthenticatedAgent
+  );
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Overwrite the agents the [[Rule]] applies to with the provided agents.
+ *
+ * @param rule The rule for which agents are set.
+ * @param agents The list of agents the rule should apply to.
+ * @returns A copy of the input rule, applying to a different set of agents.
+ * @since Unreleased
+ */
+export function setAgentInRule(rule: Rule, agent: WebId): Rule {
+  // Preserve the special agent classes authenticated and public, which we
+  // don't want to overwrite with this function.
+  // NB: This raises the question wether the named agents and agent classes should
+  // be linked to with the same predicate.
```

Comment:
It's not clear what future readers of the code are supposed to do with this comment, so I'd just leave it out, and bring it up in the relevant forums. (Which I think you've already done, right?)

```suggestion
```

Reply:
Good point, fixed in 5edb865bdaaa40d15ea658d0d23f10e3bb1e7f08